scripts/irods/test/test_all_rules.py

```diff
@@ -992,6 +995,66 @@ def do_test(entity_name, entity_type, operation, expected_output=None):
 
             do_test(data_object, 'data_object', 'add', ['atomic_pre_fired', 'atomic_post_fired'])
 
+    @unittest.skipIf(database_instance_name == 'mysql',
+                    ("Fails against databases with transaction isolation level set to REPEATABLE-READ (e.g. MySQL). "
+                     "For more details, see https://github.com/irods/irods/issues/4917"))
+    def test_msi_atomic_apply_metadata_operations_considers_group_permissions__issue_6190(self):
+        def do_test(_entity_type):
+            entity_name = os.path.join(self.admin.session_collection, 'issue_6190')
+
+            json_input = json.dumps({
+                'entity_name': entity_name,
+                'entity_type': _entity_type,
+                'operations': [
+                    {
+                        'operation': 'add',
+                        'attribute': 'issue_6190_name',
+                        'value': 'issue_6190_value',
+                        'units': 'issue_6190_units'
+                    }
+                ]
+            })
```

Comment:
I'm not sure what partial success looks like here.

As for the user vs group permissions case, that is worth capturing in an issue so that we can add it.

The atomic metadata API only touches a single object, so we should be okay for the last case.

Reply:
Oooh, you're right. Cool

Reply:
Once this PR is merged, I'll circle back and capture this in an issue.

Reply:
Created issue for this. See #6205